spec/unit/network/formats_spec.rb

```diff
@@ -394,7 +386,7 @@ def self.from_binary(data)
     it "fails intelligibly when given invalid data" do
       expect do
         json.intern(Puppet::Node, '')
-      end.to raise_error(JSON::ParserError, /A JSON text must at least contain two octets|unexpected token at ''/)
+      end.to raise_error(Puppet::Util::Json::ParseError)
```

Comment:
MultiJson returns a very cryptic error with no message at all if you pass it a "blank" string, see https://github.com/intridea/multi_json/blob/master/lib/multi_json/adapter.rb#L20. Not sure why they special cased this, the commit doesn't make that clear (without this line, we get a normal error from the JSON gem, maybe some back ends didn't handle it well?). We can either update this test to use some other kind of malformed JSON, or remove the search for a message. We might also be able to submit a patch upstream to add a message to this error.